src/run/runner/valgrind/measure.rs

```diff
@@ -42,6 +46,38 @@ lazy_static! {
     };
 }
 
+/// Creates the shell script on disk and returns the path to it.
+fn create_run_script() -> anyhow::Result<TempPath> {
+    // The command is wrapped in a shell script, which executes it in a
+    // subprocess and then writes the exit code to a file. The process will
+    // always exit with status code 0, unless valgrind fails.
+    //
+    // Args:
+    // 1. The command to execute
+    // 2. The path to the file where the exit code will be written
+    const WRAPPER_SCRIPT: &str = r#"#!/bin/sh
+            safe_run()
+            {
+               (eval $1)
+               status=$?
+               echo -n "$status" > "$2"
+            }
+
+            safe_run "$1" "$2"
```

Comment:
I think you can even remove the function here so we never quote the command
```suggestion
           (eval $1)
           status=$?
           echo -n "$status" > "$2"
```